assets/components/src/web-preview/index.js

```diff
@@ -30,95 +30,152 @@ class WebPreview extends Component {
-									>
-										<CloseIcon />
-										<span className="screen-reader-text">{ __( 'Close preview' ) }</span>
-									</Button>
-								</div>
-							</div>
-							<div className="newspack-web-preview__content">
-								{ ! loaded &&
-									<div className="newspack-web-preview__is-waiting">
-										<Waiting isLeft />
-										{ __( 'Loading...' ) }
-									</div>
-								}
-								<iframe src={ url } onLoad={ () => this.setState( { loaded: true } ) } />
-							</div>
+
+		return createPortal(
+			<div className={ classes }>
+				<div className="newspack-web-preview__interior">
+					<div className="newspack-web-preview__toolbar">
+						<div className="newspack-web-preview__toolbar-left">
+							<Button
+								onClick={ () => this.setState( { device: 'desktop' } ) }
+								isPrimary={ 'desktop' === device }
+								isSecondary={ 'desktop' !== device }
+								className="is-desktop"
+							>
+								<DesktopWindowsIcon />
+								<span className="screen-reader-text">{ __( 'Preview desktop size' ) }</span>
+							</Button>
+							<Button
+								onClick={ () => this.setState( { device: 'tablet' } ) }
+								isPrimary={ 'tablet' === device }
+								isSecondary={ 'tablet' !== device }
+								className="is-tablet"
+							>
+								<TabletAndroidIcon />
+								<span className="screen-reader-text">{ __( 'Preview tablet size' ) }</span>
+							</Button>
+							<Button
+								onClick={ () => this.setState( { device: 'phone' } ) }
+								isPrimary={ 'phone' === device }
+								isSecondary={ 'phone' !== device }
+								className="is-phone"
+							>
+								<PhoneAndroidIcon />
+								<span className="screen-reader-text">{ __( 'Preview phone size' ) }</span>
+							</Button>
+						</div>
+						<div className="newspack-web-preview__toolbar-right">
+							<Button onClick={ () => this.setState( { isPreviewVisible: false, loaded: false } ) }>
+								<CloseIcon />
+								<span className="screen-reader-text">{ __( 'Close preview' ) }</span>
+							</Button>
 						</div>
 					</div>
+					<div className="newspack-web-preview__content">
+						{ ! loaded && (
+							<div className="newspack-web-preview__is-waiting">
+								<Waiting isLeft />
+								{ __( 'Loading...' ) }
+							</div>
+						) }
+						<iframe src={ url } onLoad={ () => this.setState( { loaded: true } ) } />
+					</div>
+				</div>
+			</div>,
+			modalDOMElement
+		);
+	};
+
+	showPreview = () => {
+		this.setState( { isPreviewVisible: true } );
+	};
+
+	/**
+	 * Render.
+	 */
+	render() {
+		const {
+			label,
+			isPrimary,
+			isSecondary,
+			isTertiary,
+			isLink,
+			isLarge,
+			isSmall,
+			/**
+			 * Inversion of control - let the caller render
+			 * the button that will trigger the modal
+			 */
+			renderButton,
+		} = this.props;
+
+		return (
+			<Fragment>
+				{ renderButton ? (
+					renderButton( { showPreview: this.showPreview } )
```

Comment:
Excellent idea, I wish I had done this the first time around 😄. Do we even need the `<Button>` fallback? There are currently no uses of `WebPreview` on master yet, so we could adjust unmerged code to reflect the approach where the button UI is always provided.

Reply:
IMO a default/fallback is a good idea - we can assume the trigger will usually be the Button